java/ql/src/experimental/Security/CWE/CWE-117/LogInjection.ql

```diff
@@ -0,0 +1,67 @@
+/**
+ * @name Log Injection
+ * @description Building log entries from user-controlled data is vulnerable to
+ *              insertion of forged log entries by a malicious user.
+ * @kind path-problem
+ * @problem.severity error
+ * @precision high
+ * @id java/log-injection
+ * @tags security
+ *       external/cwe/cwe-117
+ */
+
+import java
+import DataFlow::PathGraph
+import experimental.semmle.code.java.Logging
+import semmle.code.java.dataflow.FlowSources
+
+/**
+ * A taint-tracking configuration for tracking untrusted user input used in log entries.
+ */
+private class LogInjectionConfiguration extends TaintTracking::Configuration {
+  LogInjectionConfiguration() { this = "Log Injection" }
+
+  override predicate isSource(DataFlow::Node source) { source instanceof RemoteFlowSource }
+
+  override predicate isSink(DataFlow::Node sink) {
+    sink.asExpr() = any(LoggingCall c).getALogArgument()
+  }
+
+  override predicate isSanitizer(DataFlow::Node node) {
+    node.getType() instanceof BoxedType or node.getType() instanceof PrimitiveType
+  }
+
+  override predicate isSanitizerGuard(DataFlow::BarrierGuard guard) {
+    guard instanceof StrCheckSanitizerGuard
+  }
+}
+
+/**
+ * Models any regex or equality check as a sanitizer guard.
+ * Assumes any check on the taint to be a valid sanitizing check.
+ */
+private class StrCheckSanitizerGuard extends DataFlow::BarrierGuard {
```

Comment:
Make this class also extend `MethodAccess`. You can then remove the cast from `checks`.

Reply:
If I do that I would have to override three methods unnecessarily. So I am keep it as is for now.
```
StrCheckSanitizerGuard must override ambiguous inherited method: getEnclosingCallable()
StrCheckSanitizerGuard must override ambiguous inherited method: getEnclosingStmt()
StrCheckSanitizerGuard must override ambiguous inherited method: getBasicBlock()
```